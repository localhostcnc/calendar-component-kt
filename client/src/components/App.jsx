import React from 'react';
import axios from 'axios';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      listing: null,
      bookings: null
    };
  }

  getCalendarData (cb) {
    axios.get('/calendar/1')
      .then(function (response) {
        cb(null, response);        
        console.log('Axios GET request from client Successful', response.data);
      })
      .catch(function (error) {
        cb(error);
        console.log('Axios GET request from client NOT Successful', error);
      });
  }

  componentDidMount() {
    this.getCalendarData((err, responseData) => {
      if (err) {
        console.log('Response Error: ', err);
      } else {
        var listingData = responseData.data.listing;
        var bookingData = responseData.data.bookings;
        this.setState({
          listing: listingData,
          bookings: bookingData
        });
      }
    });
  }

  render () {
    return (
      <div>
        <h1>
          Rendering App Component!
        </h1>
      </div>
    );
  }
}

export default App;