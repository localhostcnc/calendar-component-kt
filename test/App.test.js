import React from 'react';
import { shallow } from 'enzyme';

import App from '../client/src/components/App';

describe('App component', () => {
  test('Renderign the simple component', () => {
    const wrapper = shallow(
      <App />
    );
  //   const listingState = wrapper.state().listing;
  //   expect(listingState.id).toEqual(1);
  //   expect(listingState.id).to.be.an('array');

  //   // func={() => {}}
  //   // value={1}

    expect(wrapper).toMatchSnapshot();
  });

  it('has listing state of null ', () => {
    const wrapper = shallow(
      <App />
    );
    const listingState = wrapper.state().listing;
    expect(listingState).toEqual(null);
  });

  it('has a bookings state of null', () => {
    const wrapper = shallow(
      <App />
    );
    const bookingsState = wrapper.state().bookings;
    expect(bookingsState).toEqual(null);
  });
});