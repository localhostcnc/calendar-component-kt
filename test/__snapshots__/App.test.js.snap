// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Rendering the simple component 1`] = `ShallowWrapper {}`;
